Sources/AblyAssetTrackingInternal/WorkerQueue/WorkerQueue.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import AblyAssetTrackingCore
+
+/// The WorkerQueue is responsible for enqueueing ``Worker``s and executing them.
+class WorkerQueue<PropertiesType, WorkerSpecificationType> where PropertiesType: WorkerQueueProperties {
+    private var properties: PropertiesType
+    private let workingQueue: DispatchQueue
+    private let logHandler: InternalLogHandler?
+    private let workerFactory: any WorkerFactory<PropertiesType, WorkerSpecificationType>
+    private let getStoppedError: () -> Error
+
+    init(properties: PropertiesType, queue: DispatchQueue, logHandler: InternalLogHandler?, workerFactory: any WorkerFactory<PropertiesType, WorkerSpecificationType>, getStoppedError: @escaping () -> Error) {
+        self.properties = properties
+        self.workingQueue = queue
+        self.logHandler = logHandler
+        self.workerFactory = workerFactory
+        self.getStoppedError = getStoppedError
+    }
+    
+    
+    /// Enqueue worker created from passed specification for execution.
+    /// - parameters:
+    ///    - workerSpecification: ``WorkerSpecificationType`` specification of worker to be executed.
+    func enqueue(workerSpecification: WorkerSpecificationType) {
+        let worker = workerFactory.createWorker(workerSpecification: workerSpecification)
+        
+        workingQueue.async { [weak self] in
+            guard let self = self
+            else { return }
+            
+            do {
+                if self.properties.isStopped {
+                    worker.doWhenStopped(error: self.getStoppedError())
```

Comment:
It would also be useful to log before `doWork` is called and after it completes. Even better if we could (perhaps at a `verbose` level) also print the value of `properties` before and after.

Reply:
454fff79a68710148ee54a56bfc7907d65db82b5